### PR TITLE
travis: improve readability of travis.yml and log outputs

### PR DESCRIPTION
**Pull request by scravy**
This PR contains two commits which

* make the travis.yml file a bit more readable by splitting the environment variables over lines (it's a little bit more verbose but way easier to compare across forks or visually grasp in the first place),
* use `travis:fold` to fold the `autogen`, `configure`, `make`, and so forth, steps.

The `END_FOLD` shell function preserves the exit code of the statement right before.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
ACK c2f3dc0b37d3b8c442d2aa0e92db64fe3b98bedf. OK with me.

**Comment by scravy** [comment]
The build actually failed for one item in the build matrix, specifically the functional tests in `rpc_createmultisig` ( https://travis-ci.org/bitcoin/bitcoin/jobs/408882695#L3853 ).

I can't see how this would have been effected by the change to the travis file, so I'm wondering if it's like this in master and whether I should rebase.

I read in the contribution guidelines "sometimes you might be asked to rebase by a reviewer" – is rebasing in general ok? I am wondering as one of the comments refers to a commit in here and rebasing will change it's hash, so technically you're pointing into a fork of my git history which no longer should be valid :-)

**Comment by ken2812221** [comment]
utACK c2f3dc0b37d3b8c442d2aa0e92db64fe3b98bedf

File: .travis.yml [hunk 1]
```diff
@@ -64,16 +114,16 @@ script:
...
-    - if [ "$TRAVIS_EVENT_TYPE" = "cron" ]; then extended="--extended --exclude feature_pruning,feature_dbcrash"; fi
-    - if [ "$RUN_TESTS" = "true" ]; then DOCKER_EXEC test/functional/test_runner.py --combinedlogslen=4000 --coverage --quiet --failfast ${extended}; fi
+    - BEGIN_FOLD configure; DOCKER_EXEC ./configure --cache-file=../config.cache $BITCOIN_CONFIG_ALL $BITCOIN_CONFIG || ( cat config.log && false); END_FOLD
+    - BEGIN_FOLD build; DOCKER_EXEC make $MAKEJOBS $GOAL || ( echo "Build failure. Verbose build follows." && DOCKER_EXEC make $GOAL V=1 ; false ); END_FOLD
+    - BEGIN_FOLD unit-tests; if [ "$RUN_TESTS" = "true" ]; then DOCKER_EXEC LD_LIBRARY_PATH=$TRAVIS_BUILD_DIR/depends/$HOST/lib make $MAKEJOBS check VERBOSE=1; fi; END_FOLD
+    - BEGIN_FOLD extended-tests; if [ "$TRAVIS_EVENT_TYPE" = "cron" ]; then extended="--extended --exclude feature_pruning,feature_dbcrash"; fi; END_FOLD
+    - BEGIN_FOLD functional-tests; if [ "$RUN_TESTS" = "true" ]; then DOCKER_EXEC test/functional/test_runner.py --combinedlogslen=4000 --coverage --quiet --failfast ${extended}; fi; END_FOLD
```
Comment (Empact) [inline .travis.yml]: No point in including the fold if the relevant option is off, right? How about put the `BEGIN_FOLD` within the `if`s?

Comment (scravy) [reply .travis.yml]: Fixed

**Comment by Fuzzbawls** [comment]
Concept ACK

Improved readability is always nice :)

**Comment by Empact** [comment]
I notice `extended-tests` just sets up config for `test_runner` - it doesn't need a fold. How about dropping that fold, or moving the extended handling to before as below?

https://github.com/Empact/bitcoin/commit/4b858645a2cdccd5d6c9a1610974676fcba2787a
https://travis-ci.org/Empact/bitcoin/builds/409306396

**Comment by scravy** [comment]
@Empact I removed the fold for the mere configuration of the extended tests.

**Comment by MarcoFalke** [comment]
Please squash the "fold" commits

**Comment by Empact** [comment]
utACK 4639bbc, I checked that commands were faithfully refactored in 9730d3a; please squash latter 3 commits.

**Comment by scravy** [comment]
@Empact @MarcoFalke I squashed the fold-commits into one.

**Pull request merged**